typescript/scripts/build.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+# The above is the most platform-agnostic way to guarantee this script runs with Bash
+
+set -euo pipefail   # Bash "strict mode"
+
+# # ==================================================================================================
+# #                                             Constants
+# # ==================================================================================================
+script_dirpath="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+lang_root_dirpath="$(dirname "${script_dirpath}")"
+
+
+# # ==================================================================================================
+# #                                             Main Logic
+# # ==================================================================================================
+cd "${lang_root_dirpath}/mainLogic"
```

Comment:
So I try hard to avoid using `cd` in a Bash script, because it gets confusing which directory you're actually in. To get around this, I just specify full paths - e.g. I'd replace these two lines with just one:
```
find "${lang_root_dirpath}/mainLogic" -name "*.ts" -exec tsc --out "${lang_root_dirpath}/build" {} \;
```
This means there's no cd'ing, and you use the `lang_root_dirpath` variable that's already available